substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/util/Utf8.java

```diff
@@ -49,24 +63,28 @@ public static int utf8Length(String string) {
     /**
      * @param beginIndex first index that is part of the region, inclusive
      * @param endIndex index at the end of the region, exclusive
-     * @return the length in bytes of the UTF8 representation of the string region
+     * @return the length as {@code int} in bytes of the UTF8 representation of the string. Might
+     *         return a truncated size if the value does not fit into {@code int} (see JDK-8328877).
      */
     public static int utf8Length(String s, int beginIndex, int endIndex) {
+        // Based on
+        // https://github.com/openjdk/jdk/blob/jdk-24+16/src/hotspot/share/utilities/utf8.cpp#L511-L526.
         if (beginIndex < 0 || endIndex > s.length() || beginIndex > endIndex) {
             throw new StringIndexOutOfBoundsException();
         }
-        int length = 0;
-        for (int i = beginIndex; i < endIndex; i++) {
-            final int c = s.charAt(i);
-            if ((c >= 0x0001) && (c <= 0x007F)) {
-                length++;
-            } else if (c > 0x07FF) {
-                length += 3;
-            } else {
-                length += 2;
+        long result = 0;
+        for (int index = beginIndex; index < endIndex; index++) {
+            char c = s.charAt(index);
+            long sz = utf8Size(c);
+            // If the length is > INT_MAX-1 we truncate at a completed
+            // modified-UTF8 encoding. This allows for +1 to be added
+            // by the caller for NUL-termination, without overflow.
```

Comment:
Delete the comment.

Reply:
Why? It's part of the original commit?

Reply:
It's not in the JDK 21 change that this change is based on (in contrast to the 24 version):

https://github.com/openjdk/jdk21u/blob/d6e8788ca888fac7f04751e72202f10fda8cad59/src/hotspot/share/utilities/utf8.cpp#L409-L418

This repo is JDK 21 based, hence the ask.